uvicorn/config.py

```diff
@@ -123,45 +131,45 @@ def create_ssl_context(
 class Config:
     def __init__(
         self,
-        app,
-        host="127.0.0.1",
-        port=8000,
-        uds=None,
-        fd=None,
-        loop="auto",
-        http="auto",
-        ws="auto",
-        ws_max_size=16 * 1024 * 1024,
-        lifespan="auto",
-        env_file=None,
-        log_config=LOGGING_CONFIG,
-        log_level=None,
-        access_log=True,
-        use_colors=None,
-        interface="auto",
-        debug=False,
-        reload=False,
-        reload_dirs=None,
-        reload_delay=None,
-        workers=None,
-        proxy_headers=True,
-        forwarded_allow_ips=None,
-        root_path="",
-        limit_concurrency=None,
-        limit_max_requests=None,
-        backlog=2048,
-        timeout_keep_alive=5,
-        timeout_notify=30,
-        callback_notify=None,
-        ssl_keyfile=None,
-        ssl_certfile=None,
-        ssl_keyfile_password=None,
-        ssl_version=SSL_PROTOCOL_VERSION,
-        ssl_cert_reqs=ssl.CERT_NONE,
-        ssl_ca_certs=None,
-        ssl_ciphers="TLSv1",
-        headers=None,
-        factory=False,
+        app: Union[ASGIApplication, Awaitable, Callable, str],
+        host: str = "127.0.0.1",
+        port: int = 8000,
+        uds: Optional[str] = None,
+        fd: Optional[int] = None,
+        loop: str = "auto",
+        http: Union[Type[asyncio.Protocol], str] = "auto",
+        ws: Union[Type[asyncio.Protocol], str] = "auto",
+        ws_max_size: int = 16 * 1024 * 1024,
+        lifespan: str = "auto",
+        env_file: Optional[Union[Path, str]] = None,
+        log_config: Optional[Union[dict, str]] = LOGGING_CONFIG,
```

Comment:
can be a file too

Reply:
Yes, but the file path would be coming in as a string, right? The log config logic assumes string in some places, like where `.endswith()` is used:

https://github.com/encode/uvicorn/blob/87da6cf4082cd28306bdb78d7d42552d131cc179/uvicorn/config.py#L252-L259

Reply:
yes you're right, I had in minc click.Path could give you a Path object but it's not the case